src/kernel-install/90-loaderentry.install.in

```diff
@@ -48,8 +48,29 @@ else
     ENTRY_DIR="${ENTRY_DIR_ABS#"$BOOT_MNT"}"
 fi
 
+KERNEL_DEST="$ENTRY_DIR_ABS/linux"
```

Comment:
hmm, i'd really prefer if this specific commit could be done in a separate commit, i think the gc logic is relatively independent of the kernel-install additions.

Reply:
I can factor out those variables of course. Not sure the plugin support itself is too useful without reference counting though. kernel-install itself would rm -rf the whole directory on remove calls. So whatever renaming of kernel and initrd a plugin would do, a remove call removes all of them.

Reply:
only variables: https://github.com/systemd/systemd/pull/25992